code/modules/mob/living/carbon/alien/humanoid/queen.dm

```diff
@@ -10,9 +10,74 @@
 	layer = LARGE_MOB_LAYER //above most mobs, but below speechbubbles
 	pressure_resistance = 200 //Because big, stompy xenos should not be blown around like paper.
 	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab/xeno = 20, /obj/item/stack/sheet/animalhide/xeno = 3)
+	gps_name = "Hissing Signal"
 
+	var/life_ticks_to_wait = 10 //life does some expensive things, so we only want to do the calculation occasionally
 	var/alt_inhands_file = 'icons/mob/alienqueen.dmi'
 
+/mob/living/carbon/alien/humanoid/royal/queen/Login()
+	..()
+	SSshuttle.registerHostileEnvironment(src)
+
+/mob/living/carbon/alien/humanoid/royal/queen/Logout()
+	..()
+	addtimer(CALLBACK(src, .proc/logoutcheck), 12 SECONDS)//when logging out, we don't want to spam announcements with leaving and reconnecting.
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
+
+/mob/living/carbon/alien/humanoid/royal/queen/death()
+	..()
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
+
+/mob/living/carbon/alien/humanoid/royal/queen/proc/logoutcheck()
+	if(!client)
+		SSshuttle.clearHostileEnvironment(src) //the queen has officially gone braindead, so we can allow the shuttle to leave
+
+/mob/living/carbon/alien/humanoid/royal/queen/Life()
+	..()
+	if(client)
+		if(life_ticks_to_wait)
+			life_ticks_to_wait--
+			return
+		life_ticks_to_wait = initial(life_ticks_to_wait)
+		var/living_humans = 0
+		var/total_humans = length(GLOB.human_list)
+		for(var/H in GLOB.human_list)
+			var/mob/living/carbon/human/human = H
+			if(!human.client || human.stat == DEAD )
+				continue
+			living_humans++
+
+		if(living_humans < total_humans/10 && !nuking && !neutralized)
+			INVOKE_ASYNC(src, .proc/nuke_it_from_orbit)
+
+/mob/living/carbon/alien/humanoid/royal/queen/proc/nuke_it_from_orbit()
+	nuking = TRUE
+	sleep(50)

```

Comment:
Don't use sleep. Use timers

Reply:
In this context why is it better to use timers instead of sleep?

Reply:
sleep is a byond inbuilt we don't use (except in very specific cases), we generally prefer using our own timer system.